### PR TITLE
Add comment to complex regex for matching invalid XML

### PR DESCRIPTION
**Pull request by jfsoul**
## What does this change?
After using the `stripInvalidXMLCharacters` method and struggling to understand the regex, I have added a (very long 😰) comment to the pattern for clarity.  Too much?

## What is the value of this and can you measure success?
Easier to understand code.

### CODE REVIEW

File: common/app/common/TrailsToRss.scala [hunk 1]
```diff
@@ -20,6 +20,14 @@ import scala.collection.JavaConverters._
 
 object TrailsToRss extends implicits.Collections {
 
+  /* 
+    This regex pattern matches all invalid XML characters (see https://www.w3.org/TR/xml/#charsets)
+    by specifying individual and ranges of valid ones in a negated set.  The final range \\u10000-\\u10FFFF
+    is intended to match the supplementary character set, but I believe it is invalid java regex.
+    It ought to be changed to \\x{10000}-\\x{10FFFF} but that produces unexpected behaviour which I suspect
```
Comment (NataliaLKB) [inline common/app/common/TrailsToRss.scala]: Do you have a reference for where this bug has been reported/talked about?

**Review by NataliaLKB — APPROVED** [review]
Thanks for adding this comment @jfsoul. It is really helpful for others wanting to understand what this is doing. I made one little comment/question, but otherwise its great!

File: common/app/common/TrailsToRss.scala (hunk 1 above)
Comment (jfsoul) [inline common/app/common/TrailsToRss.scala]: I didn't when I wrote this, but I filed the bug now and have added the link! 👇

Comment (NataliaLKB) [inline common/app/common/TrailsToRss.scala]: Thanks!

**Pull request merged**